windows/gui-source/Kolibri/Kolibri.cpp

```diff
@@ -44,6 +44,28 @@ wchar_t * getStr(int strId) {
 	return finalStr;
 }
 
+char * getKolibriLinkAddress() {
+	/*
+	This will return the current Kolibri running link url.
+	It will fetch the HTTP_PORT value from options.ini config file.
+	*/
+	char * kolibriHomeEnv = getenv("KOLIBRI_HOME");
+	if (kolibriHomeEnv == NULL) {
+		kolibriHomeEnv = "";
+	}
+	char * configName = "\\options.ini";
+	char * filePath = joinChr(kolibriHomeEnv, configName);
+
+	const size_t fileStrSize = strlen(filePath) + 1;
+	wchar_t* configPath = new wchar_t[fileStrSize];
+	mbstowcs(configPath, filePath, fileStrSize);
+	UINT httpPort = GetPrivateProfileInt(L"Deployment", L"HTTP_PORT", 8080, configPath);

```

Comment:
I also agree that this is the wrong approach, we cannot assume that the ini file is the source of truth here. The `kolibri status` command provides this info in a very robust, cross-platform way. @benjaoming could you advise on how to use the output of that command in this instance?

Reply:
Yeah, the 8080 is use by default if `.ini` not define the port. I think I should also consider the port from environment variable. the --port argument I think is optional because it should be run at the command prompt not in the kolibri application.